src/decoder.rs

```diff
@@ -129,20 +159,28 @@ pub fn decode_regular(rsm: RawSourceMap) -> Result<SourceMap> {
 
     let names = rsm.names.unwrap_or_default();
     let sources = rsm.sources.unwrap_or_default();
+    let range_mappings = rsm.range_mappings.unwrap_or_default();
     let mappings = rsm.mappings.unwrap_or_default();
     let allocation_size = mappings.matches(&[',', ';'][..]).count() + 10;
     let mut tokens = Vec::with_capacity(allocation_size);
+    let mut range_tokens = vec![];
 
     let mut nums = Vec::with_capacity(6);
 
-    for (dst_line, line) in mappings.split(';').enumerate() {
+    for (dst_line, (line, rmi_str)) in mappings
+        .split(';')
+        .zip(range_mappings.split(';').chain(std::iter::repeat("")))
+        .enumerate()
+    {
         if line.is_empty() {
             continue;
         }
 
         dst_col = 0;
 
-        for segment in line.split(',') {
+        let rmi = decode_rmi(rmi_str)?;
```

Comment:
no matter if you end up with `bitvec` or any other type, please make sure to reuse that in this iteration instead of allocating/returning a new one.
Similar to how `parse_vlq_segment_into` takes a `&mut Vec` to amortize allocations.